src/ArrayUtils.f90

```diff
@@ -971,4 +972,21 @@ PURE FUNCTION findUpBound_1DReal(r,pos,delta,incl,tol) RESULT(val)
       ENDIF
     ENDFUNCTION findUpBound_1DReal
 !
+!-------------------------------------------------------------------------------
+!> @brief This function checks if an index is within the allocated domain of a
+!>        1D array of unspecified type. Currently can only be used for 1D arrays
+!>        until assumed-shape polymorphic arrays are supported.
+!> @param array the 1D array to be checked
+!> @param idx the index to check
+!>
+    PURE FUNCTION boundCheck(array,idx) RESULT(res)
+      CLASS(*),INTENT(IN) :: array(:)
```

Comment:
Did not realize you could use this...

I figured this would get implemented as a type bound generic.